tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/VariableService.java

```diff
@@ -175,17 +195,30 @@ public void persistTaskVariables(
               var.getName(),
               var.getValue(),
               tasklistProperties.getImporter().getVariableSizeThreshold()));
+
+      // Add Variables added to a Task as a Process Variablee
+      finalVariablesListViewMap.put(
+          var.getName(),
+          VariableListViewEntity.createFrom(
+              task.getTenantId(),
+              task.getFlowNodeInstanceId() + "-" + var.getName(),
+              var.getName(),
+              var.getValue(),
+              task.getFlowNodeInstanceId(),
+              tasklistProperties.getImporter().getVariableSizeThreshold(),
+              new ListViewJoinRelation("taskVariable", taskFlowNodeInstanceId)));
     }
+    listViewStore.persistTaskVariables(finalVariablesListViewMap.values());
```

Comment:
Does this apply for variables of tasks not migrated to ListView?
We may persist the variable in ListView and not the task itself
Or is the non-migrated task persisted in ListView by the importer?

Reply:
Does this apply for variables of tasks not migrated to ListView? 
R: At the end it will persist only the variables that will become process variables, once we will remove the task variables from job workers
We are not persisting the Task, only the Variables form the task. But the Process Variables are still necessary to be there.

Or is the non-migrated task persisted in ListView by the importer?
The importer will bring all variables (once it is not possible identify variables coming from V1 or V2). But after task competition, only input variables from Job workers will be removed. 
Non-Migrated tasks wont be imported (Job Workers)